apps/costcontrol/js/settings/limitdialog.js

```diff
@@ -56,6 +59,21 @@ function dataLimitConfigurer(guiWidget, settings, viewManager, widgetRoot) {
     );
   }
 
+  function checkDataLimit(dataLimit) {
+    CostControl.getInstance(function _onCostControl(costcontrol) {
+      var requestObj = {
+        type: 'datausage'
+      };
+      costcontrol.request(requestObj, function _onDataStats(result) {
+        var stats = result.data.mobile.total;
+        if (stats > dataLimit) {
+          var message = _('data-limit-notification-text2');
```

Comment:
Notice Juwei said [you should show the same message as in the notification](https://bugzilla.mozilla.org/show_bug.cgi?id=1139811#c33) which includes the part of `XX of data used`. You're missing this part right now.